### PR TITLE
Forfeit Game Feature; JS Drag Feature; Castling Feature

### PR DESCRIPTION
**Pull request by msarit**
Hello team. Apologies for not creating pull requests sooner 🙈 

This PR contains the following:

**(1) Forfeit Game feature**
I added 4 enums for the game state: `active`, `complete`, `forfeited` and `stalemate`. Then I updated the game model and controller. It's RSpec tests aren't passing though, even though the functionality is working.

**(2) Making Pieces Draggable**
Added JQuery for this in the drag.js file. Couldn't get the drag to reverse if the move isnt valid; Ilya mentioned this may be because `valid_move?` hasn't been called when pieces move.

**(3) Castling Feature**
This is incomplete - I need the `in_check?` method, a way to determine that neither King nor Rook has moved, and how to pass `rook_x` and `rook_y` in `castle!(rook_x, rook_y)` from the game_view to the piece_controller. I'll go over the last two items with my mentor tonight.

😄 😄 😄

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -128,5 +114,40 @@ def move_to!(new_x,new_y)
       return "ERROR! Cannot move there; occupied by a friendly piece"
     end
   end
+
+  # = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
+
+
+  def can_castle?(rook_x, rook_y)
```
Comment (ilyakrasnov) [inline app/models/piece.rb]: @msarit  
should that not be on the king? line of thought: "can this king castle?"
or maybe even on the game.rb? line of thought: "can black or white player castle?"

**Comment by ilyakrasnov** [comment]
@msarit added one comment.. otherwise looks great.. 👍 

please pull current master and try merging into your feature branch (you'll need to resolve conflicts).. then push the branch again to github in order to be able to merge without conflicts...

File: app/models/piece.rb (hunk 1 above)
Comment (msarit) [reply app/models/piece.rb]: @ilyakrasnov it makes more sense to me to have `can_castle?(x,y)` in the king model 😄 I'll make the changes.

For determining whether black or white player can castle, I included this logic in the game_showpage itself.

**Comment by Linda091** [comment]
Reviewed the code with Arit this AM, so +1 from me!

**Pull request merged**